truss/truss_handle.py

```diff
@@ -397,11 +396,38 @@ def training_docker_build_setup(self, build_dir: Optional[Path] = None):
 
     def add_python_requirement(self, python_requirement: str):
         """Add a python requirement to truss model's config."""
-        self._update_config(
-            lambda conf: replace(
-                conf, requirements=[*conf.requirements, python_requirement]
+
+        # Raise error if user forgets to put "=="
+        try:
+            requirement_name, requirement_version = python_requirement.split("==")
```

Comment:
These requirements follow the full pip spec. You can easily parse the reqs using `pkg_resources.Requirement.parse`, e.g. here:
https://github.com/basetenlabs/truss/blob/c182964a7a7f9a82ba986091df1e09c2a3c32fde/truss/patch/calc_patch.py#L217

We can check if the req already exists and if it's different, similar to here, but using pkg_resources.

Reply:
Really appreciate your contribution, this should be a simple change, and then I'm happy to merge it.

Reply:
My pleasure, sounds good! working on it!

Reply:
I made it work the same way by using pkg_resources now, which now it doesn't require any split hardcoding, it should work for many general cases

Reply:
I tried with different operations than '==', and it was updating the version for them as well, this is now operation agnostic since it just updates the version by replacing the version through Requirements.specs

Reply:
It is also able to change the operation as well without duplicating it. It updates the existing package operation, and if the version also is changed at the same time, updates the version too together with the operation